packages/flp-config-sub-generator/src/app/index.ts

```diff
@@ -0,0 +1,266 @@
+    isExtensionInstalled,
+    getHostEnvironment,
+    hostEnvironment,
+    YUI_EXTENSION_ID,
+    YUI_MIN_VER_FILES_GENERATED_MSG
+} from '@sap-ux/fiori-generator-shared';
+import { withCondition } from '@sap-ux/inquirer-common';
+import { generatorTitle, i18nKeySubTitle, i18nKeyTitle } from '../utils/constants';
+import { t } from '../utils';
+import { getYuiNavStep } from '../utils/prompts';
+import { EventName } from '../telemetryEvents';
+import type { FLPConfigAnswers } from '@sap-ux/flp-config-inquirer';
+import type { YeomanEnvironment, VSCodeInstance } from '@sap-ux/fiori-generator-shared';
+import type { Manifest, ManifestNamespace } from '@sap-ux/project-access';
+import type { FlpConfigOptions } from './types';
+import type { Answers, Question } from 'inquirer';
+
+/**
+ * FLP config generator adds an inbound navigation config to an existing manifest.json.
+ */
+export default class extends DeploymentGenerator {
+    private readonly appWizard: AppWizard;
+    private readonly vscode?: VSCodeInstance;
+    private readonly launchFlpConfigAsSubGenerator: boolean;
+    private readonly existingApp: boolean;
+    private readonly appRootPath: string;
+    private readonly prompts: Prompts;
+    private answers: FLPConfigAnswers;
+    private abort = false;
+    private manifest: Partial<Manifest>;
+    private manifestPath: string;
+    public options: FlpConfigOptions;
+
+    setPromptsCallback: (fn: object) => void;
+
+    /**
+     * Constructor for FLP config generator.
+     *
+     * @param args - arguments passed to the generator
+     * @param opts - options passed to the generator
+     */
+    constructor(args: string | string[], opts: FlpConfigOptions) {
+        super(args, opts);
+
+        this.appWizard = opts.appWizard ?? AppWizard.create(opts);
+        this.vscode = opts.vscode;
+        this.launchFlpConfigAsSubGenerator = opts.launchFlpConfigAsSubGenerator ?? false;
+        this.appRootPath = opts?.data?.appRootPath ?? opts?.appRootPath ?? this.destinationRoot();
+        this.options = opts;
+
+        // If launched standalone, add nav steps
+        if (!this.launchFlpConfigAsSubGenerator) {
+            this.appWizard.setHeaderTitle(generatorTitle);
+            this.prompts = new Prompts(getYuiNavStep(this.appRootPath));
+            this.setPromptsCallback = (fn): void => {
+                if (this.prompts) {
+                    this.prompts.setCallback(fn);
+                }
+            };
+            this.existingApp = true; // generator is launched standalone so app must exist
+        }
+    }
+
+    public async initializing(): Promise<void> {
+        if ((this.env as unknown as YeomanEnvironment).conflicter) {
+            (this.env as unknown as YeomanEnvironment).conflicter.force = this.options.force ?? true;
+        }
+
+        await TelemetryHelper.initTelemetrySettings({
+            consumerModule: {
+                name: '@sap/generator-fiori-deployment:flp-config',
```

Comment:
Should not be referencing tools suite modules from open source. Just use the actual generator name directly. `this.generatorName`  or the current package name.

Reply:
Updated